File: Grundlagen/OOP/AG3-Klassen/Telefonliste/person.php
<?php
class person {
 
    private $id;
    private $vorname;
    private $nachname;
    private $telefonnummer;
    
 
    
    public function __construct($id, $nachname, $vorname, $telefonnummer) {
        
        $this -> id=$id;
        $this -> vorname = $vorname;
        $this -> nachname = $nachname;
        $this -> telefonnummer=$telefonnummer;
    }
    
    
    public function setID($id) {
        $this->id=$id;
    }
    
    public function getID() {
        return $this-> id;
    }
    
    public function setVorname($vorname) {
        $this -> vorname=$vorname;
    }
    
    public function getVorname(){
        return $this -> vorname;
    }
    
    
    public function setNachname($nachname) {
        $this -> nachname=nachname;
    }
    
    public function getNachname(){
        return $this -> nachname;
    }
    
    public function setTelefonnummer($telefonnummer) {
        $this-> telefonnummer=$telefonnummer;
    }
    
    public function getTelefonnummer() {
        return $this -> telefonnummer;
    }
    
}


    
?>

File: InterfacesGh/KlasseTabelle/Person.java

public class Person {

	private String nachname;
	private String vorname;
	private String telefonnummer;
	
	Person () {}

	public Person(String nachname, String vorname, String telefonnummer) {
		super();
		this.nachname = nachname;
		this.vorname = vorname;
		this.telefonnummer = telefonnummer;
	}

	public String getNachname() {
		return nachname;
	}

	public void setNachname(String nachname) {
		this.nachname = nachname;
	}

	public String getVorname() {
		return vorname;
	}

	public void setVorname(String vorname) {
		this.vorname = vorname;
	}

	public String getTelefonnummer() {
		return telefonnummer;
	}

	public void setTelefonnummer(String telefonnummer) {
		this.telefonnummer = telefonnummer;
	}

	
	

	
	
	@Override
	public String toString() {
		return "Person [nachname=" + nachname + ", vorname=" + vorname + ", telefonnummer=" + telefonnummer + "]";
	}

	@Override
	public boolean equals (Object obj) {
		if (!(obj instanceof Person))
		{
			return false;
		}
		Person pr = (Person) obj;
		if ( (this.nachname == pr.nachname) && (this.vorname==pr.vorname) )
			return true;

		return false;
	}
	
	
	@Override
	public int hashCode() {
		return vorname.length() * nachname.length();
		
	}
	
}
	
